lib/tictactoe.rb

```diff
@@ -1,58 +1,82 @@
 # frozen_string_literal: true
 
 class TicTacToe
-  def initialize(io:)
-    @board = Array.new(9, ' ')
+  def initialize(io:, board:)
+    @board = board
     @current_token = 'X'
     @io = io
   end
 
   def render_board
-    board_output = <<~BOARD_RENDER 
-    #{@board[0]}|#{@board[1]}|#{@board[2]}
+    board_output = <<~BOARD_RENDER
+    #{@board.get_info_at_position(position: 0)}|#{@board.get_info_at_position(position: 1)}|#{@board.get_info_at_position(position: 2)}
     ------
-    #{@board[3]}|#{@board[4]}|#{@board[5]}
+    #{@board.get_info_at_position(position: 3)}|#{@board.get_info_at_position(position: 4)}|#{@board.get_info_at_position(position: 5)}
     ------
-    #{@board[6]}|#{@board[7]}|#{@board[8]}
+    #{@board.get_info_at_position(position: 6)}|#{@board.get_info_at_position(position: 7)}|#{@board.get_info_at_position(position: 8)}
     BOARD_RENDER
-    io.puts board_output
-    board
+    io.output_to_user board_output
+    board_output
```

Comment:
I think this return value may only be used in your tests, and doesn't seem necessary from a logic standpoint. I'm pretty eager to delete unused code, because I'm sold on the idea that less code is easier to maintain, and code that isn't being used is especially hard to maintain (because the app running correctly isn't proof that the dead code works correctly). If you're only testing the return value, and not that the board output is passed to `io.output_to_user`, that strikes me as a gap in your test coverage.